query-engine/connectors/sql-query-connector/src/database/operations/write.rs

```diff
@@ -12,16 +16,82 @@ use std::{
 use tracing::log::trace;
 use user_facing_errors::query_engine::DatabaseConstraint;
 
+async fn generate_id(
+    conn: &dyn QueryExt,
+    conn_info: ConnectionInfo,
+    model: &ModelRef,
+    args: WriteArgs,
+    trace_id: Option<String>,
+) -> crate::Result<(Option<SelectionResult>, bool)> {
+    let pk = model.primary_identifier();
+
+    // fast path if it's not mysql
+    if conn_info.sql_family() != SqlFamily::Mysql {
+        return Ok((args.as_record_projection(pk.into()), false));
+    }
+
+    // Go to all the values and generate a select statement with the correct mysql function
+    let (pk_select, generated) = pk
+        .clone()
+        .selections()
+        .filter_map(|field| match field {
+            SelectedField::Scalar(x) if x.default_value.is_some() => x
+                .default_value
+                .clone()
+                .unwrap()
+                .dbgenerated_func()
+                .map(|func| (field.db_name().to_string(), func)),
+            _ => None,
+        })
+        .fold((Select::default(), false), |(query, generated), value| {
+            let alias = value.0;
+            let func = value.1.to_lowercase();
+            match func.trim() {
+                "(uuid())" => (query.value(native_uuid().alias(alias)), true),
+                "(uuid_to_bin(uuid()))" => (query.value(uuid_to_bin().alias(alias)), true),
+                _ => (query, generated),
+            }
+        });
+
+    // db generate values only if needed
+    if generated {
+        let pk_select = pk_select.add_trace_id(trace_id);
+        let pk_result = conn.query(pk_select.into()).await?;
+        let result = try_convert(&pk.into(), pk_result)?;
+
+        Ok((Some(result), true))
+    } else {
+        Ok((args.as_record_projection(pk.into()), false))
+    }
+}
+
 /// Create a single record to the database defined in `conn`, resulting into a
 /// `RecordProjection` as an identifier pointing to the just-created record.
 #[tracing::instrument(skip(conn, model, args))]
 pub async fn create_record(
     conn: &dyn QueryExt,
+    conn_info: ConnectionInfo,
     model: &ModelRef,
     args: WriteArgs,
     trace_id: Option<String>,
 ) -> crate::Result<SelectionResult> {
-    let (insert, returned_id) = write::create_record(model, args, trace_id);
+    let (returned_id, generated) = generate_id(conn, conn_info, model, args.clone(), trace_id.clone()).await?;
+
+    // Append db generated values if required
+    let args = match returned_id {
+        Some(ref pk) if generated => {
+            let mut args = args.clone();
+            for (field, value) in pk.pairs.iter() {
+                let field = DatasourceFieldName(field.prisma_name().to_string());
```

Comment:
There should be a converter for the db name struct already (`field.into`).

Reply:
Nope, there is none. You cannot convert a `SelectedField` into `DatasourceFieldName` and that struct requires a `&str`, in this particular case `field.db_name().into()` is converting to `&str` which is the same as previous.

Though, you found a bug in my implementation, I should use `db_name` and not `prisma_name`. Changed to `into` as a way to say thanks for finding the bug :cook: